xapian-core/weight/tfidfweight.cc

```diff
@@ -211,42 +268,41 @@ TfIdfWeight::get_wdfn(Xapian::termcount wdf, Xapian::termcount doclen,
 	    return num / den;
 	}
 	default:
-	    AssertEq(c, 'n');
 	    return wdf;
     }
 }
 
 double
-TfIdfWeight::get_idfn(char c) const
+TfIdfWeight::get_idfn(idfn_type idf_norm_) const
 {
     Xapian::doccount termfreq = 1;
-    if (c != 'n') termfreq = get_termfreq();
+    if (idf_norm_ != idfn_type::NONE) termfreq = get_termfreq();
     double N = 1.0;
-    if (c != 'n' && c != 'f') N = get_collection_size();
-    switch (c) {
-	case 'n':
+    if ((idf_norm_ == idfn_type::PROB) || (idf_norm_ == idfn_type::SQUARE) ||
+	(idf_norm_ == idfn_type::TFIDF) || (idf_norm_ == idfn_type::PIVOTED))
```

Comment:
There's no need to put extra parentheses around each equality test, and the code seems harder to read that way.

Why did you change the sense of the tests here?  Previously we checked the norm wasn't one of two, but now we check it is one of the other four - twice as many comparisons!

Reply:
This change was supposed to be after I add more normalization. For now I have restored the previous version.
Thanks for pointing it out.